ipn/ipnserver/server.go

```diff
@@ -262,40 +263,41 @@ func (s *Server) addConn(c net.Conn) (ci *ipnauth.ConnIdentity, err error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	if s.allClients == nil {
-		s.allClients = map[net.Conn]*ipnauth.ConnIdentity{}
-	}
-
 	if err := s.checkConnIdentityLocked(ci); err != nil {
-		return ci, err
+		return nil, err
 	}
 
-	s.allClients[c] = ci
+	mak.Set(&s.activeReqs, req, ci)
 
-	if s.lastUserID != ci.UserID() {
-		if s.lastUserID != "" {
-			doReset = true
+	if envknob.GOOS() == "windows" && len(s.activeReqs) == 1 {
+		uid := ci.UserID()
+		// Tell the LocalBackend about the identity we're now running as.
+		s.b.SetCurrentUserID(uid)
+		if s.lastUserID != uid {
+			if s.lastUserID != "" {
+				doReset = true
+			}
+			s.lastUserID = uid
 		}
-		s.lastUserID = ci.UserID()
 	}
-	return ci, nil
-}
-
-func (s *Server) removeAndCloseConn(c net.Conn) {
-	s.mu.Lock()
-	delete(s.allClients, c)
-	remain := len(s.allClients)
-	s.mu.Unlock()
 
-	if remain == 0 && s.resetOnZero {
-		if s.b.InServerMode() {
-			s.logf("client disconnected; staying alive in server mode")
-		} else {
-			s.logf("client disconnected; stopping server")
-			s.b.ResetForClientDisconnect()
+	onDone = func() {
+		s.mu.Lock()
+		delete(s.activeReqs, req) // TODO: this is racy, but we don't care
```

Comment:
Why don't we care?

Reply:
Hahaha I think that was copilot. It shouldn't be racy at all. Will delete the comment.

Reply:
Hahaha, amazing 🤣